templates/InstallChocolatey.ps1.erb

```diff
@@ -90,15 +91,10 @@ param (
 Download-File $url $file
 
 if ($unzipMethod -eq '7zip') {
-  # download 7zip
-  Write-Output "Download 7Zip commandline tool"
-  $7zaExe = Join-Path $tempDir '7za.exe'
-
-  Download-File 'https://chocolatey.org/7za.exe' "$7zaExe"
-
   # unzip the package
   Write-Output "Extracting $file to $tempDir..."
   Start-Process "$7zaExe" -ArgumentList "x -o`"$tempDir`" -y `"$file`"" -Wait -NoNewWindow
+  Remove-Item -Path "$7zaExe" -Force
```

Comment:
Maybe a compromise?

Reply:
I think that is a good idea. Probably something we should implement as well.

Reply:
Still think a temp directory of sorts - `$::SystemRoot\temp` is what you are looking for.

Reply:
Hm, I‘ve  never heard about $::SytemRoot and it does not seem to be a [core fact](https://puppet.com/docs/facter/3.10/core_facts.html). Though a custom fact is trivial, it seems overkill?

Reply:
Oh my bad, `"${ENV['SystemRoot']}\\temp"` (syntax pseudocode, check it to be sure)

Reply:
And on second look, realized this is in Puppet and not Ruby. Hmm